src/Analyzers/CSharp/Analyzers/UseIndexOrRangeOperator/CSharpUseRangeOperatorDiagnosticAnalyzer.cs

```diff
@@ -130,6 +131,14 @@ protected override void InitializeWorker(AnalysisContext context)
                 return null;
             }
 
+            var theIndexer = GetIndexer(targetMethod.ContainingType, infoCache.RangeType, targetMethod.ContainingType);
+            // If the slice-like method is being written to and returns a reference, we need to make sure that
+            // the range method to substitute also returns a reference that can be written to
+            if (invocation.Syntax.IsLeftSideOfAnyAssignExpression() && invocation.TargetMethod.ReturnsByRef == true && theIndexer != null && theIndexer.ReturnsByRef == false)
```

Comment:
```suggestion
            if (indexer != null && indexer.ReturnsByRef !=invocation.TargetMethod.ReturnsByRef &&  invocation.Syntax.IsLeftSideOfAnyAssignExpression())
```

Reply:
although, my preference is simply that we update ComputeMemberInfo to only return an indexer if it matches the byref semantics of hte slice method.  i.e. the code to update would be hereish:

```
                // https://github.com/dotnet/csharplang/blob/master/proposals/csharp-8.0/ranges.md#implicit-range-support
                if (sliceLikeMethod.ReturnType.Equals(containingType))
                {
                    // it's a method like:  MyType MyType.Get(int start, int length).  Look for an
                    // indexer like  `MyType MyType.this[Range range]`.
                    var indexer = GetIndexer(containingType, RangeType, containingType);
```

GetIndexer should probalby take a ReturnsByRef and only provides indexers that match.

Reply:
I tried quite a few different approaches to accessing the same information through ComputeMemberInfo but I wasn't able to get the IsLeftSideOfAnyAssignExpression information without adding some kind of argument to the method so I decided to keep it as is.